server/group/group.js

```diff
@@ -0,0 +1,150 @@
+const router = require('express').Router();
+const { Group, GroupMember, User } = require('../db/index');
+
+router.post('/newgroup', async (req, res, next) => {
+  try {
+    const user = await User.findByPk(req.session.userId);
+    const newGroup = Group.create({...req.body.group});
+  }
+  catch (error) {
+    next(error);
+  }
+});
+
+router.get('/all/:next', async (req, res, next) => {
+  try {
+    let filters = {};
+    let order = {};
+    if (req.body.filter) {
+      for(let f of req.body.filter) {
+	if(f.hasOwnProperty('zipcode')) {
+	  // All all custom filters here
+	  filters.zipcode = f.zipcode;
+	}
+      }
+      for(let o of req.body.order) {
+	if(o.hasOwnProperty('createdAt')) {
+	  // All all custom order here
+	  filters.createdAt = o.createdAt;
+	}
+      }
+    }
+    const groupsAll = await Group.findAll(
+      [{ offset: req.params.next * 25, limit: 25,
+	 where: { ...filters },
+	 order: [...order],
+       }
+      ]);
+    res.send(groupsAll);
+  }
+  catch (error) {
+    next(error);
+  }
+});
+
+// Gets all users for a specific group
+router.get('/groupusers', async (req, res, next) => {
+  try {
+    const group = Group.findByPk(req.body.groupId);
+    const groupUsers = group.getUsers();
+    res.send(groupUsers);
+  }
+  catch (error) {
+    next(error);
+  }
+});
+
+// Gets all groups for a specific user
+router.get('/user', async (req, res, next) => {
+  try{
+    // I think this is right, I'm going to have to test it.
+    const user = await User.findByPk(req.session.userId);
+    const userGroups = await user.getGroups();
+    res.send(userGroups);
+  }
+  catch (error) {
+    next(error);
+  }
+});
+
+router.post('/create', async (req, res, next) => {
```

Comment:
Should this be here? It looks like we already have a new group post route above?